core/src/verify.rs

```diff
@@ -454,6 +487,22 @@ impl CommitSequenceVerifier {
         self.total_commits.push(commit.clone());
         Ok(())
     }
+
+    /// To verify that the newly added previous_block_hash field effectively prevents replay attacks
+    /// Check the integrity of the chain of agendas when processing new agendas
+    pub fn verify_agenda_integrity(
+        new_agenda: &Agenda,
+        prev_header_hash: &Hash256,
+    ) -> Result<(), Error> {
+        if new_agenda.previous_block_hash != *prev_header_hash {
+            return Err(Error::InvalidArgument(format!(
+                "invalid agenda previous_block_hash: expected {}, got {}",
+                prev_header_hash, new_agenda.previous_block_hash
+            )));
```

Comment:
How about fixing here to return Err() and fix the error message in apply_commit like this one?

Reply:
reflected like:

```rust
pub fn verify_agenda_integrity(
        new_agenda: &Agenda,
        prev_header_hash: &Hash256,
    ) -> Result<(), Error> {
        if new_agenda.previous_block_hash != *prev_header_hash {
            return Err(Error::AgendaIntegrityError(
                prev_header_hash.to_string(),
                new_agenda.previous_block_hash.to_string(),
            ));
        }

        Ok(())
    }

                if CommitSequenceVerifier::verify_agenda_integrity(
                    agenda,
                    &self.header.to_hash256(),
                )
                    .is_err()
                {
                    return Err(Error::AgendaIntegrityError(
                        agenda.previous_block_hash.to_string(),
                        self.header.to_hash256().to_string(),
                    ));
                }
                self.phase = Phase::Agenda {
                    agenda: agenda.clone(),
                };
```